nuxeo-core/nuxeo-core/src/main/java/org/nuxeo/ecm/core/api/AbstractSession.java

```diff
@@ -2031,6 +2036,87 @@ protected Object getDataModelField(DocumentRef docRef, String schema, String fie
         return null;
     }
 
+    @Override
+    public void makeRecord(DocumentRef docRef) {
+        Document doc = resolveReference(docRef);
+        if (isRecord(docRef)) {
+            // already a record, don't do anything
+            return;
+        }
+        checkPermission(doc, CAN_MAKE_RECORD);
+        DocumentModel docModel = readModel(doc);
+        notifyEvent(DocumentEventTypes.BEFORE_MAKE_RECORD, docModel, null, null, null, true, false);
+        doc.makeRecord();
+        docModel = readModel(doc);
+        notifyEvent(DocumentEventTypes.AFTER_MAKE_RECORD, docModel, null, null, null, true, false);
+    }
+
+    @Override
+    public boolean isRecord(DocumentRef docRef) {
+        Document doc = resolveReference(docRef);
+        checkPermission(doc, READ);
+        return doc.isRecord();
+    }
+
+    @Override
+    public void setRetentionUntil(DocumentRef docRef, Calendar retainUntil) throws PropertyException {
+        Document doc = resolveReference(docRef);
+        if (!isRecord(docRef)) {
+            throw new PropertyException("Document is not a record");
+        }
+        Calendar current = doc.getRetentionUntil();
+        if (Objects.equals(current, retainUntil)) {
+            // unchanged, don't do anything
+            return;
+        }
+        checkPermission(doc, CAN_SET_RETENTION);
+        Map<String, Serializable> options = new HashMap<>();
+        options.put(CoreEventConstants.RETENTION_UNTIL, retainUntil);
+        if (retainUntil != null) {
+            options.put("comment", retainUntil.toInstant().toString()); // for audit
```

Comment:
What do you think about using `DateUtils` for calendar serialization?

Reply:
`DateUtils` is not available in 10.10